src/components/layouts/loginDialog.tsx

```diff
@@ -0,0 +1,121 @@
+import { Field } from "@/components/ui/field";
+import {
+  DialogBody,
+  DialogCloseTrigger,
+  DialogContent,
+  DialogTitle,
+  DialogRoot,
+  DialogHeader,
+} from "@/components/ui/dialog";
+import { Stack, Input, Text } from "@chakra-ui/react";
+import { LuLogIn } from "react-icons/lu";
+import { Button } from "@/components/ui/button";
+import useMainStore from "@/hooks/main/store";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import useAuthStore from "@/contexts/auth/store";
+import clsx from "clsx";
+import { SignInDto } from "@/contexts/auth/types/signIn";
+import { SignInSchema } from "@/contexts/auth/schemas/signIn";
+import { SignInField } from "@/contexts/auth/fields/signIn";
+
+const LoginDialog = () => {
+  const { isLoginDialogOpen, setIsLoginDialogOpen, setIsRegisterDialogOpen } =
+    useMainStore();
+  const { isLoading, signIn } = useAuthStore();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<SignInDto>({
+    resolver: yupResolver(SignInSchema),
+  });
+
+  const onSubmit = handleSubmit((data) => {
+    signIn({
+      email: data.email,
+      password: data.password,
+    });
+  });
+
+  return (
+    <DialogRoot
+      lazyMount
+      open={isLoginDialogOpen}
+      onOpenChange={() => setIsLoginDialogOpen(false)}
+    >
+      <DialogContent className="bg-slate-500 text-gray-100">
+        <DialogHeader>
+          <DialogTitle
+            textAlign="center"
+            fontSize="2xl"
+            fontWeight="bold"
+            mb={4}
+          >
+            Login
+          </DialogTitle>
+        </DialogHeader>
+        <DialogBody as="form" onSubmit={onSubmit}>
+          <Stack wordSpacing={4}>
+            {SignInField.map((field) => (
+              <Field
+                key={field.name}
+                label={field.label}
+                invalid={!!errors[field.name]}
+                errorText={errors[field.name]?.message}
+              >
+                <Input
+                  {...register(field.name)}
+                  type={field.type}
+                  placeholder={field.placeholder}
+                  className={clsx(
+                    "rounded-lg border-2 p-4 focus:ring-2 bg-gray-100 text-lg text-gray-800 placeholder-gray-400 appearance-none",
+                    !errors[field.name]
+                      ? "focus:ring-blue-500"
+                      : "focus:ring-red-500",
```

Comment:
is this have a possibility to be a repetitive code,
maybe u can create a handler function that return that css, so it gonna be look like this:
```js
clsx("css.....", handlerErrorRing(!errors[field.name]));
```
or something more simple